core/jazz_events-handler/index.js

```diff
@@ -14,125 +14,135 @@
-
-				if (interestedEvents.indexOf(record.kinesis.partitionKey) !== -1) {
-					var params = JSON.parse(payload);
-					params.ReturnConsumedCapacity = "TOTAL";
-					params.TableName = config.EVENT_TABLE;
-
-					dynamodb.putItem(params, function (err, data) {
-						if (err) {
-							logger.error('Unable to store event in events table, message: ' + JSON.stringify(err));
-							failedEvents.push({
-								"sequence_id": sequenceNumber,
-								"event": payload,
-								"failure_code": failureCodes.DB_ERROR_1.code,
-								"failure_message": failureCodes.DB_ERROR_1.message
-							});
-							innerCallback({
-								"db_error": "Unable to store event in events table"
-							});
-
-						} else {
-							logger.info('Stored interested event in table');
-							processedEvents.push({
-								"sequence_id": sequenceNumber,
-								"event": payload,
-								"failure_code": null,
-								"failure_message": null
-							});
-							innerCallback(null, {
-								"message": "successfully processed interested event"
-							});
-						}
-
-					});
-				} else {
-					processedEvents.push({
-						"sequence_id": sequenceNumber,
-						"event": payload,
-						"failure_code": null,
-						"failure_message": null
-					});
-					logger.info('Skipping storage of un-interested event');
-					innerCallback(null, {
-						"message": "successfully processed un-interested event"
-					});
-				}
-
-			}, function (err) {
-				callback();
-			});
-
-		}
-
-	}, function (err, results) {
-		if (err) {
-			return logger.error('Error inside events handler ' + JSON.stringify(err));
-		}
-
-		cb(null, {
-			"processed_events": processedEvents.length,
-			"failed_events": failedEvents.length
-		});
-	});
-
+var processedEvents = [];
+var failedEvents = [];
+var failureCodes = fcodes();
+
+function handler(event, context, cb) {
+  //Initializations
+  let config = configModule.getConfig(event, context);
+
+  exportable.getEvents(config)
+    .then(res => exportable.processRecords(config, event, res))
+    .then(() => { 
+      return cb(null, { "processed_events": processedEvents.length, "failed_events": failedEvents.length }); 
+    })
+    .catch(err => {
+      return logger.error('Error inside events handler ' + JSON.stringify(err));
+    });
 };
+
+function getEvents(config) {
+  return new Promise((resolve, reject) => {
+    const docClient = new AWS.DynamoDB.DocumentClient();
+    let interestedEvents = [];
+    let params = {
+      TableName: config.EVENT_NAME_TABLE,
+      ProjectionExpression: "EVENT_NAME",
+      Limit: 1000,
+      ReturnConsumedCapacity: "TOTAL"
+    };
+
+    let scanExecute = function () {
+      docClient.scan(params, function (err, result) {
+        if (err) {
+          return reject({ "db_error": "Unable to scan Event Names table to fetch interested events" });
+        } else {
+          //interestedEvents = interestedEvents.concat(result.Items);
+          for (i = 0; i < result.Items.length; i++) {
```

Comment:
please use map, filter and reduce. For loops are slower in performance in general.
This can be written like this in just one line.
const interestedEvents = result.Items.map(i => i.EVENT_NAME);
Please update in other places as well.

Reply:
no sir.
https://ryanpcmcquen.org/javascript/2015/10/25/map-vs-foreach-vs-for.html

Reply:
Interesting.  This might be a small use case where performance might not be a big consideration but the basic fundamental behind for is that it will mutate and map does not. For multiple reasons i strongly recommend using map.
Just go through "Benefits of map of for loops" 
https://medium.com/@ExplosionPills/map-vs-for-loop-2b4ce659fb03